pkg/virt-controller/watch/vmi.go

```diff
@@ -200,6 +203,59 @@ func (e *syncErrorImpl) Reason() string {
 	return e.reason
 }
 
+type cidsMap struct {
+	mu   sync.Mutex
+	cids map[string]uint32
+}
+
+// Sync loads the allocated CIDs from VMIs.
+func (m *cidsMap) Sync(vmis []*virtv1.VirtualMachineInstance) {
+	m.mu.Lock()
+	defer m.mu.Unlock()
+	for _, vmi := range vmis {
+		if vmi.Status.VsockCID == nil {
+			continue
+		}
+		key := controller.VirtualMachineInstanceKey(vmi)
+		m.cids[key] = *vmi.Status.VsockCID
+	}
+}
+
+// Allocate select a new CID and set it to the status of the given VMI.
+func (m *cidsMap) Allocate(vmi *virtv1.VirtualMachineInstance) {
+	m.mu.Lock()
+	defer m.mu.Unlock()
+	key := controller.VirtualMachineInstanceKey(vmi)
+	if cid, exist := m.cids[key]; exist {
+		vmi.Status.VsockCID = &cid
+		return
+	}
+	reverseMap := make(map[uint32]string)
+	var maxCID uint32
+	for key, cid := range m.cids {
```

Comment:
let's just always fill both the `cids` map and the `reverseMap` instead of rebuilding the `reverseMap` every time.

Reply:
Done